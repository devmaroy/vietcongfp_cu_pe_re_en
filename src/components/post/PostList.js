import React from 'react';
import PropTypes from 'prop-types';
import Pagination from "react-js-pagination";
import PostItem from './PostItem';

class PostList extends React.Component {
    state = {
        activePage: 1,
        postsPerPage: 5
    }

    handlePageChange = ( pageNumber ) => {
        this.setState( () => ( { activePage: pageNumber } ) );
    }

    render() {
        const { activePage, postsPerPage } = this.state;
        const { posts } = this.props;

        return (
            <div>
                {
                    posts.length === 0 ? (
                        <p>There are no posts.</p>
                    ) : (
                        <div>
                            <div className="posts">
                                {
                                    posts
                                        .slice( ( activePage - 1 ) * postsPerPage, ( postsPerPage * activePage ) )
                                        .map( ( post ) => <PostItem key={ post.id } post={ post } /> )
                                }
                            </div>
                            {
                                posts.length > postsPerPage && (
                                    <div className="site-pagination">
                                        <Pagination
                                            activePage={ this.state.activePage }
                                            itemsCountPerPage={ postsPerPage }
                                            totalItemsCount={ posts.length }
                                            pageRangeDisplayed={ 5 }
                                            onChange={ this.handlePageChange }
                                        />
                                    </div>
                                )
                            }
                        </div>
                    )
                }
            </div>
        );
    };
};

PostList.propTypes = {
    posts: PropTypes.array.isRequired
};

export default PostList;