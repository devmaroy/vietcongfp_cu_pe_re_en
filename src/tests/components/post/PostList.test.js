import React from 'react';
import { shallow } from 'enzyme';
import PostList from '../../../components/post/PostList';
import posts from '../../fixtures/posts';

test( 'should render PostList with empty message', () => {
    const wrapper = shallow(
        <PostList
            posts={ [] }
        />
    );

    expect( wrapper ).toMatchSnapshot();
});

test( 'should render PostList correctly with data', () => {
    const wrapper = shallow(
        <PostList 
            posts={ posts }
        />
    );

    expect( wrapper ).toMatchSnapshot();
});

test( 'should show Pagination if length of posts is more than postsPerPage in state', () => {
    const wrapper = shallow(
        <PostList
            posts={ posts }
        />
    );

    wrapper.setState({
        postsPerPage: 1
    });

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find( '.site-pagination' ).exists() ).toBeTruthy();
});

test( 'should not show Pagination if length of posts is less than postsPerPage in state', () => {
    const wrapper = shallow(
        <PostList
            posts={ posts }
        />
    );

    wrapper.setState({
        postsPerPage: 20
    });

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find( '.site-pagination' ).exists() ).toBeFalsy();
});