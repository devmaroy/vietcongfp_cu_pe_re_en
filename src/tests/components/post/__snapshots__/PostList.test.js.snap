// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not show Pagination if length of posts is less than postsPerPage in state 1`] = `
<div>
  <div>
    <div
      className="posts"
    >
      <Connect(PostItem)
        key="1"
        post={
          Object {
            "body": "Even after 10 years we have new patch!",
            "createdAt": 1000,
            "id": "1",
            "name": "JoshTh",
            "title": "New patch",
          }
        }
      />
      <Connect(PostItem)
        key="2"
        post={
          Object {
            "body": "Our gallery was improved and we have now new better pictures",
            "createdAt": 1200,
            "id": "2",
            "name": "JoshTh",
            "title": "Gallery updated",
          }
        }
      />
      <Connect(PostItem)
        key="3"
        post={
          Object {
            "body": "Yes! We need more members in our team. Feel free to contact us.",
            "createdAt": 1500,
            "id": "3",
            "name": "Patrick",
            "title": "We are hiring!",
          }
        }
      />
      <Connect(PostItem)
        key="4"
        post={
          Object {
            "body": "As you can see we have new bug in our lovely game!",
            "createdAt": 2000,
            "id": "4",
            "name": "Pete088",
            "title": "Weird bug found",
          }
        }
      />
      <Connect(PostItem)
        key="5"
        post={
          Object {
            "body": "We have brand new section on our page with custom contact form",
            "createdAt": 2300,
            "id": "5",
            "name": "JoshTh",
            "title": "Contact form added to our website",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`should render PostList correctly with data 1`] = `
<div>
  <div>
    <div
      className="posts"
    >
      <Connect(PostItem)
        key="1"
        post={
          Object {
            "body": "Even after 10 years we have new patch!",
            "createdAt": 1000,
            "id": "1",
            "name": "JoshTh",
            "title": "New patch",
          }
        }
      />
      <Connect(PostItem)
        key="2"
        post={
          Object {
            "body": "Our gallery was improved and we have now new better pictures",
            "createdAt": 1200,
            "id": "2",
            "name": "JoshTh",
            "title": "Gallery updated",
          }
        }
      />
      <Connect(PostItem)
        key="3"
        post={
          Object {
            "body": "Yes! We need more members in our team. Feel free to contact us.",
            "createdAt": 1500,
            "id": "3",
            "name": "Patrick",
            "title": "We are hiring!",
          }
        }
      />
      <Connect(PostItem)
        key="4"
        post={
          Object {
            "body": "As you can see we have new bug in our lovely game!",
            "createdAt": 2000,
            "id": "4",
            "name": "Pete088",
            "title": "Weird bug found",
          }
        }
      />
      <Connect(PostItem)
        key="5"
        post={
          Object {
            "body": "We have brand new section on our page with custom contact form",
            "createdAt": 2300,
            "id": "5",
            "name": "JoshTh",
            "title": "Contact form added to our website",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`should render PostList with empty message 1`] = `
<div>
  <p>
    There are no posts.
  </p>
</div>
`;

exports[`should show Pagination if length of posts is more than postsPerPage in state 1`] = `
<div>
  <div>
    <div
      className="posts"
    >
      <Connect(PostItem)
        key="1"
        post={
          Object {
            "body": "Even after 10 years we have new patch!",
            "createdAt": 1000,
            "id": "1",
            "name": "JoshTh",
            "title": "New patch",
          }
        }
      />
    </div>
    <div
      className="site-pagination"
    >
      <a
        activePage={1}
        firstPageText="«"
        getPageUrl={[Function]}
        hideFirstLastPages={false}
        innerClass="pagination"
        itemsCountPerPage={1}
        lastPageText="»"
        nextPageText="⟩"
        onChange={[Function]}
        pageRangeDisplayed={5}
        prevPageText="⟨"
        totalItemsCount={5}
      />
    </div>
  </div>
</div>
`;
